packages/store/src/patterns/encodeKey.js

```diff
@@ -38,7 +40,17 @@ const asBits = new BigUint64Array(asNumber.buffer);
 // encoding whose lexicographic sort order is the same as the numeric sort order
 // of the corresponding numbers.
 
+// TODO Choose the same canonical NaN encoding that cosmWasm and ewasm chose.
+const CanonicalNaN = 'ffff8000000000000';
+
+// Normalize -0 to 0
+
 const numberToDBEntryKey = n => {
+  if (is(n, -0)) {
+    n = 0;
+  } else if (is(n, NaN)) {
```

Comment:
Out of curiosity, why not use `Number.isNaN`?

I rarely remember the equality rules for `NaN`.

Reply:
No reason except that I was already using `is`.